src/rgw/driver/rados/rgw_pubsub.cc

```diff
@@ -449,13 +450,28 @@ int RGWPubSub::get_topics(rgw_pubsub_topics *result) const
   return read_topics(result, nullptr);
 }
 
-int RGWPubSub::Bucket::read_topics(rgw_pubsub_bucket_topics *result, RGWObjVersionTracker *objv_tracker) const
+int RGWPubSub::Bucket::read_topics(const DoutPrefixProvider *dpp, rgw_pubsub_bucket_topics *result, RGWObjVersionTracker *objv_tracker) const
 {
+  //read from cache
+  auto cache = ps.store->getRados()->get_topic_cache();
+  const std::string key = bucket_meta_obj.pool.to_str() + bucket_meta_obj.oid;
+  if (auto e = cache->find(key)) {
+    *result = e->info;
+    return 0;
+  }
   int ret = ps.read(bucket_meta_obj, result, objv_tracker);
   if (ret < 0 && ret != -ENOENT) {
     ldout(ps.store->ctx(), 1) << "ERROR: failed to read bucket topics info: ret=" << ret << dendl;
     return ret;
   }
+  if (result) {
+    rgw_cache_entry_info cache_info;
+    pubsub_bucket_topics_entry e;
+    e.info = *result;
+    if (!cache->put(dpp, ps.store->getRados()->svc.cache, key, &e, { &cache_info })) {
+      ldout(ps.store->ctx(), 10) << "couldn't put bucket topics cache entry" << dendl;
```

Comment:
since you added a `dpp` argument to this function, you can use it for this log statement:  `ldpp_dout(dpp, 10) <<`